### DISCUSSION TITLE
Disable template cache in KC 17 (Quarkus)

### DISCUSSION THREAD

**Discussion by Toilal** [keycloak/keycloak#10319] [Archive]
How can I disable the template cache for template development with Keycloak 17 (Quarkus) ?

I was using the following startup script `disable-theme-cache.cli` in previous version, but obviously it doesn't work anymore.

```
embed-server --std-out=echo --server-config=standalone-ha.xml
/subsystem=keycloak-server/theme=defaults/:write-attribute(name=cacheThemes,value=false)
/subsystem=keycloak-server/theme=defaults/:write-attribute(name=cacheTemplates,value=false)
/subsystem=keycloak-server/theme=defaults/:write-attribute(name=staticMaxAge,value=-1)
stop-embedded-server
```

**Comment by dasniko** [comment]
If you start Keycloak in dev mode (`start-dev`), the template cache should be disabled.

**Reply by DGuhr** [reply]
Right @dasniko 

Apart from that, when you want to disable it elsewhere, it's 

```
spi-theme-cache-themes=false
spi-theme-cache-templates=false
spi-theme-static-max-age=-1
```

Reference: internal [keycloak.conf](https://github.com/keycloak/keycloak/blob/febb447919044943241ad97200821c7da5099cb3/quarkus/runtime/src/main/resources/META-INF/keycloak.conf)

**Reply by Toilal** [reply]
Thanks. Those settings are missing from the docs page of server settings.

**Reply by stianst** [reply]
@DGuhr I think we need to consider making some SPI provider config (like themes) proper config options. We may also want to prioritise the ability to view/list SPI provider properties.

**Reply by kanguran** [reply]
As I understand disabling templates caching might be useful in Developers local environment.
How Enable Templates caching on the server, aren't those HTML caches enabled be default ?

**Reply by eduardolucioac** [reply]
The "/opt/keycloak/themes/README.md" file says...

"""
[...]
When running the server in development mode using `start-dev`, themes are not cached so that you can easily work on them without a need to restart
[...]
"""

KC 19.0.3 😎

**Comment by Ragatzino** [comment]
Does this work with gzip formatted caches ? I wanted to keep some caching server side, but lower the time so i tried:

spi-theme-static-max-age=1000 

And this doesn't seem to work ..

**Reply by Ragatzino** [reply]
My goal is simply to have caching in production for gz files and normal files in themes. But in the documentation I only see informations about "developer" mode, i would like to have some kind of production usages and configuration of spi-theme-cache for production, with redeployment without having to either rename, use some sort of "hack" in changing the cli on the redeploy and then putting cache back up. Just something that would do all this at once..
